### PR TITLE
[share] Add share_tizen package

### PR DESCRIPTION
**Pull request by bwikbs**

### CODE REVIEW

**Comment by bwikbs** [comment]
This plugin require platform apps such as ``email`` to work properly  and there are  no such apps in public images. The only thing that works right now is the ``Message`` app in the mobile profile.

**Comment by swift-kim** [comment]
Can't you install apps like e-mail (from app store) on a watch or TV?

**Comment by bwikbs** [comment]
It has not been confirmed whether such an app exists in app store. Testing of this app was done with an app attached to an old commercial image.

**Comment by bbrto21** [comment]
> It has not been confirmed whether such an app exists in app store. Testing of this app was done with an app attached to an old commercial image.

If there is no problem with the basic operation, I think it's enough to merge. I don't think we need to consider the lack of an app ecosystem of tizen and no apps that respond to this plugin in the store

**Review by swift-kim — COMMENTED** [review]
The Message and E-mail apps are pre-installed on my Galaxy Watch and TM1 by default. The share_tizen plugin works with both apps properly on TM1, but only works with the Message app on Galaxy Watch.

| | Galaxy Watch | TM1 |
|-|:-:|:-:|
| SMS (only text) | O | O |
| E-mail (text with a subject) | AppControl error (below) | O |
| File | ? | ? |


```
I/ConsoleMessage( 5642): [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: PlatformException(-17825759, APP_CONTROL - The application is not found, null, null)
I/ConsoleMessage( 5642): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:582)
I/ConsoleMessage( 5642): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:159)
I/ConsoleMessage( 5642): <asynchronous suspension>
I/ConsoleMessage( 5642): #2      DemoAppState._onShare (package:share_example/main.dart:118)
```

Also an integration test is missing which may be added later.

File: packages/share/LICENSE [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+Copyright (c) 2017 The Chromium Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (bwikbs) [reply packages/share/LICENSE]: done.

File: packages/share/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Share Tizen plugin
+
+The Tizen implementation of [`share`](https://github.com/flutter/plugins/tree/master/packages/share).
+
+## Usage
+
+To use this plugin, add `share` and `share_tizen` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).
```
[the remark this answers is not in the document]
Comment (bwikbs) [reply packages/share/README.md]: done.

File: packages/share/example/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# share_example
+
+Demonstrates how to use the share plugin.
+
+## Getting Started
+
+For help getting started with Flutter, view our online
+[documentation](https://flutter.dev/).
```
[the remark this answers is not in the document]
Comment (bwikbs) [reply packages/share/example/README.md]: done.

File: packages/share/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+To share one or multiple files invoke the static `shareFiles` method anywhere in your Dart code. Optionally you can also pass in `text` and `subject`.
+``` dart
+Share.shareFiles(['${directory.path}/image.jpg'], text: 'Great picture');
+Share.shareFiles(['${directory.path}/image1.jpg', '${directory.path}/image2.jpg']);
+```
+
+## Notice
+To use this plugin, you need to declare privileges in tizen-manifest.xml of your application.
+``` xml
+<privilege>http://tizen.org/privilege/appmanager.launch</privilege>
```
[the remark this answers is not in the document]
Comment (bwikbs) [reply packages/share/README.md]: done.

**Comment by swift-kim** [comment]
I made some additional changes. Please check out https://github.com/swift-kim/plugins/commits/pr/32. (You can add me as a collaborator in your repo if you want me to commit the change.)

- Remove examples in README that are not supported by share_tizen. Also add more detailed description on the limitations.
- Remove `subject: 'title'` from the integration test since it's not supported
- Check if `subject` is null to avoid errors when it's null

**Comment by bwikbs** [comment]
> I made some additional changes. Please check out https://github.com/swift-kim/plugins/commits/pr/32. (You can add me as a collaborator in your repo if you want me to commit the change.)

I think there was a mistake while doing rebase. 
Now, you are collaborator of that repo.

**Review by swift-kim — APPROVED** [review]

**Pull request merged**